### DISCUSSION TITLE
How to get the symbolic font name from a text block?

### DISCUSSION THREAD

**Discussion by dxu** [pymupdf/PyMuPDF#3666] [Q&A]
How can i get the symbolic font name from a text block?

It seems like there is no straightforward way to do this - the text span's `font` attribute only gives me the base font name (not the symbolic id), and the same thing happens with `get_texttrace`. 

Are there other ways to accurately map a text's font back to the original font objects returned from `get_page_fonts`? (the edge case I'm trying to handle is when the font name is the same but there are multiple references)

**Comment by JorjMcKie** [comment]
The reason here is that the logic behind `page.get_fonts()` is PDF-specific. Other documents do not have that (in any similar way).
Whereas text extraction works the same way for all supported documents, from PDF over XPS to e-book formats.
So a tedious way to make some matching between the two approaches is necessary.
This however comes across other occasional problems: The font name returned in text extraction may or may not be the same as we see it in `get_fonts()`. Hyphens may be missing or be spaces, the name in text extraction has a size limit of 31 characters - which is not the case in `get_fonts()`, etc.

So the answer to your qeustion
> Are there other ways to accurately map a text's font back to the original font objects returned from get_page_fonts?

is **_no_**.
